src/pal/src/thread/thread.cpp

```diff
@@ -573,7 +568,22 @@ CorUnix::InternalCreateThread(
         palError = ERROR_INVALID_PARAMETER;
         goto EXIT;
     }
-    
+
+    if (dwStackSize != 0)
+    {
+        // Some systems require the stack size to be aligned to the page size
+        DWORD alignedStackSize = static_cast<DWORD>(ALIGN_UP(dwStackSize, OS_PAGE_SIZE));
+        if (alignedStackSize < dwStackSize)
+        {
+            // When coming here from the public API surface, the incoming value is originally a nonnegative signed int32, so
+            // this shouldn't happen
+            ASSERT("Couldn't align the requested stack size (%u) to the page size\n", dwStackSize);
```

Comment:
Since the pthread_attr_setstacksize accepts size_t parameter, I would do this slightly differently so that the problem you assert for cannot happen. I would create size_t alignedStackSize first, then compute it from dwStackSize cast to size_t and then use it in all of the rest of the function code instead of dwStackSize. That way, the condition you are asserting for cannot happen.